File: eds-starter-onemany/src/main/resources/static/app/model/inventory/InventoryDetail.js
Ext.define("Starter.model.inventory.InventoryDetail",
{
  extend : "Starter.model.inventory.Base",
  requires : [ "Ext.data.identifier.Negative", "Ext.data.proxy.Direct", "Ext.data.validator.Length" ],
  identifier : "negative",
  fields : [ {
    name : "itemId",
    type : "integer",
    allowNull : true
  }, {
    name : "quantity",
    type : "string",
    validators : [ {
      type : "length",
      min : 0,
      max : 255
    } ]
  }, {
    name : "useItem",
    type : "boolean"
  }, {
    name : "inventoryHeaderId",
    type : "integer",
    allowNull : true,
    reference : {
      parent : "InventoryHeader"
    }
  }, {
    name : "id",
    type : "integer",
    allowNull : true,
    convert : null
  } ],
  proxy : {
    type : "direct",
    api : {
      read : "detailService.read",
      create : "detailService.create",
      update : "detailService.update",
      destroy : "detailService.destroy"
    },
    writer : {
      writeAllFields : true
    }
  }
});

File: eds-starter-onemany/src/main/java/ch/rasc/eds/starter/entity/inventory/InventoryDetail.java
package ch.rasc.eds.starter.entity.inventory;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.validation.constraints.Size;

import org.hibernate.validator.constraints.NotBlank;

import com.fasterxml.jackson.annotation.JsonIgnore;

/**
 * Created by Administrator on 26/08/2015.
 */

import ch.rasc.eds.starter.entity.setup.Item;
import ch.rasc.edsutil.entity.AbstractPersistable;
import ch.rasc.extclassgenerator.Model;
import ch.rasc.extclassgenerator.ModelField;
import ch.rasc.extclassgenerator.ReferenceConfig;

@Entity
@Table(name = "InventoryDetail")
@Model(extend = "Starter.model.inventory.Base",value = "Starter.model.inventory.InventoryDetail",readMethod = "detailService.read",
        createMethod = "detailService.create",updateMethod = "detailService.update",
        destroyMethod = "detailService.destroy",identifier = "negative")
public class InventoryDetail extends  AbstractPersistable{

    @ManyToOne
    @JoinColumn(name = "itemId")
    @JsonIgnore
    private Item item;


    @Transient
    @ModelField(useNull = true)
    private Long itemId;

    @NotBlank(message = "{fieldrequired}")
    @Size(max = 255)
    private String quantity;

    @ManyToOne
    @JoinColumn(name = "inventoryHeaderId")
    @JsonIgnore
    private InventoryHeader inventoryHeader;
    @Transient
    private boolean useItem;

    @Transient
    @ModelField(reference = @ReferenceConfig(parent = "InventoryHeader"),useNull = true)
    private Long inventoryHeaderId;

    public Item getItem() {
        return this.item;
    }

    public void setItem(Item item) {
        this.item = item;
    }

    public Long getItemId() {
        return this.itemId;
    }

    public void setItemId(Long itemId) {
        this.itemId = itemId;
    }

    public String getQuantity() {
        return this.quantity;
    }

    public void setQuantity(String quantity) {
        this.quantity = quantity;
    }

    public InventoryHeader getInventoryHeader() {
        return this.inventoryHeader;
    }

    public void setInventoryHeader(InventoryHeader inventoryHeader) {
        this.inventoryHeader = inventoryHeader;
    }

    public Long getInventoryHeaderId() {
        return this.inventoryHeaderId;
    }

    public void setInventoryHeaderId(Long inventoryHeaderId) {
        this.inventoryHeaderId = inventoryHeaderId;
    }

	public boolean isUseItem() {
		return this.useItem;
	}

	public void setUseItem(boolean useItem) {
		this.useItem = useItem;
	}

}
